packages/@react-stately/collections/docs/collections.mdx

```diff
@@ -0,0 +1,387 @@
+
+## Dynamic collections
+
+Static collections are great when the items never changes, but how about dynamic data? A **dynamic collection**
+is a collection that is based on data, for example from an API. In addition, it may change over time as
+items are added, updated, or removed from the collection by a user.
+
+React Spectrum implements a JSX-based interface for dynamic collections, which maps over your data and
+applies a function for each item to render it. The following example shows how a collection can be 
+rendered based on dynamic data, stored in React state.
+
+```tsx
+let [animals, setAnimals] = useState([
+  {id: 1, name: 'Aardvark'},
+  {id: 2, name: 'Kangaroo'},
+  {id: 3, name: 'Snake'}
+]);
+
+<ListBox items={animals}>
+  {item => <Item>{item.name}</Item>}
+</ListBox>
+```
+
+As you can see, the items are passed to the `items` prop of the top-level component, which iterates over each 
+item and calls the function passed as children to the component. The item object is passed to the function, which
+returns an &lt;<TypeLink links={docs.links} type={docs.exports.Item} />&gt;.
+
+### Unique keys
+
+All items in a collection must have a unique key or id, which is used to determine what items in the collection changed
+when updates occur. By default, React Spectrum looks for an `id` or `key` property on each item, which is often returned 
+from a database. You can also specify a custom key on each item that should be used as the unique key for the item with 
+the `itemKey` prop. For example, if all animals in the example had a unique `name` property, then `itemKey` could be set 
+to `"name"` to use it as the unique key.
+
+```tsx
+let [animals, setAnimals] = useState([
+  {name: 'Aardvark'},
+  {name: 'Kangaroo'},
+  {name: 'Snake'}
+]);
+
+<ListBox items={animals} itemKey="name">
+  {item => <Item>{item.name}</Item>}
+</ListBox>
+```
+
+### Why not array map?
+
+If you're an experienced React developer, you may be wondering why we didn't use `animals.map` in this example.
+In fact, you can do this if you want and it will work, but it will not be as performant.
+
+```tsx
+let [animals, setAnimals] = useState([
+  {name: 'Aardvark'},
+  {name: 'Kangaroo'},
+  {name: 'Snake'}
+]);
+
+<ListBox>
+  {animals.map(item => 
+    <Item key={item.name}>{item.name}</Item>
+  )}
+</ListBox>
+```
+
+Using the `items` prop and providing a render function allows React Spectrum to automatically cache the results
+of rendering each item and avoid re-rendering all items in the collection when only one of them changes. This 
+has big performance benefits for large collections.
+
+### Updating data
+
+When you need to update the data to add, remove, or change an item, you can do so using a standard React state update.
+
+**IMPORTANT: all items passed to a collection component must be immutable. Changing a property on an item, or calling 
+`array.push()` or other mutating methods will not work as expected. To make this easier, see the 
```

Comment:
To help manage this? To help make this easier? ... see the useListData hook and its helper/helpful methods.
I dunno about this one at all, I mostly want to sell the hook a little more.